#### T3 — 행정동 미시 입지 데이터 수집

| API | 엔드포인트 | 분기 필터 | 처리 방식 |
|---|---|---|---|
| 3.1 매출 | VwsmAdstrdSelngW | 정상 | 분기별 수집 |
| 3.2 유동인구 | VwsmAdstrdFlpopW | 무시 | 정적 처리 |
| 3.3 점포수 | VwsmAdstrdStorW | 정상 | 분기별 수집 |
| 3.4 직장인구 | VwsmAdstrdWrcPopltnW | 정상 | 분기별 수집 |
| 3.5 집객시설 | VwsmAdstrdFcltyW | 무시 | 정적 처리 |
| 3.6 변화지표 | VwsmAdstrdIxQq | 무시 | 정적 처리 |

In [2]:
import os, json, time
from pathlib import Path
from urllib.request import urlopen
from urllib.error import URLError, HTTPError
import numpy as np
import pandas as pd

In [4]:
# 경로 설정
PROJECT_ROOT = Path('..') if Path.cwd().name == 'notebooks' else Path('.')
RAW_DIR = PROJECT_ROOT / 'data' / 'raw'
EXT_DIR = PROJECT_ROOT / 'data' / 'external'
RAW_DIR.mkdir(parents=True, exist_ok=True)
EXT_DIR.mkdir(parents=True, exist_ok=True)

# API 키 로드 (.env 파일)
def load_env(path):
    env = {}
    if not path.exists():
        print('.env 파일 없음')
        return env
    for line in path.read_text(encoding='utf-8-sig').splitlines():
        line = line.strip()
        if not line or line.startswith('#') or '=' not in line:
            continue
        k, v = line.split('=', 1)
        env[k.strip()] = v.strip().strip('"').strip("'")
    return env


env = load_env(PROJECT_ROOT / '.env')
SEOUL_API_KEY = env.get('SEOUL_API_KEY', '')
assert SEOUL_API_KEY, 'SEOUL_API_KEY 없음'
print(f'API 키 로드 완료: {SEOUL_API_KEY[:6]}...')

# 수집 분기 (2020Q1 ~ 2025Q4)
BASE_URL = 'http://openapi.seoul.go.kr:8088'
QUARTERS = [f'{y}{q}' for y in range(2020, 2026) for q in range(1, 5)]
print(f'수집 분기: {QUARTERS[0]} ~ {QUARTERS[-1]} (총 {len(QUARTERS)}개)')

API 키 로드 완료: 414f43...
수집 분기: 20201 ~ 20254 (총 24개)


#### API 호출 함수

In [5]:
def fetch_one_page(endpoint, start, end, extra=''):
    suffix = f'/{extra}' if extra else ''
    url = f'{BASE_URL}/{SEOUL_API_KEY}/json/{endpoint}/{start}/{end}{suffix}'
    for attempt in range(3):
        try:
            with urlopen(url, timeout=30) as resp:
                payload = json.loads(resp.read().decode('utf-8'))
            block = payload.get(endpoint, {})
            result = block.get('RESULT', {})
            if result and result.get('CODE') not in ('INFO-000', None):
                return []
            return block.get('row', [])
        except (URLError, HTTPError, json.JSONDecodeError) as e:
            if attempt == 2:
                print(f'호출 실패 ({endpoint}, {extra}): {e}')
                return []
            time.sleep(1.5)
    return []

def fetch_all(endpoint, extra='', max_rows=50000):
    rows = []
    for start in range(1, max_rows + 1, 1000):
        end = start + 999
        page = fetch_one_page(endpoint, start, end, extra)
        if not page:
            break
        rows.extend(page)
        if len(page) < 1000:
            break
    return rows

print('함수 정의 완료')

함수 정의 완료


#### 자치구 매핑 파일 수집

In [10]:
# 매핑 파일 수집
MAPPING_PATH = RAW_DIR / 'TbgisTrdarRelm.csv'

if MAPPING_PATH.exists():
    df_map = pd.read_csv(MAPPING_PATH, dtype=str)
    print(f'캐시 사용: {len(df_map)}건')
else:
    rows = fetch_all('TbgisTrdarRelm', max_rows=2000)
    df_map = pd.DataFrame(rows)
    df_map.to_csv(MAPPING_PATH, index=False, encoding='utf-8-sig')
    print(f'수집 완료: {len(df_map)}건')

print(sorted(df_map['SIGNGU_CD_NM'].unique()))

캐시 사용: 1650건
['강남구', '강동구', '강북구', '강서구', '관악구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문구', '동작구', '마포구', '서대문구', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포구', '용산구', '은평구', '종로구', '중구', '중랑구']


#### 분기 필터 검증(1000행)

In [11]:
verify_apis = {
    '3.1 매출':    'VwsmAdstrdSelngW',
    '3.2 유동인구': 'VwsmAdstrdFlpopW',
    '3.3 점포수':  'VwsmAdstrdStorW',
    '3.4 직장인구': 'VwsmAdstrdWrcPopltnW',
    '3.5 집객시설': 'VwsmAdstrdFcltyW',
    '3.6 변화지표': 'VwsmAdstrdIxQq',
}
test_quarters = ['20231', '20232', '20233']

for name, endpoint in verify_apis.items():
    print(f"\n[{name}] 분기 필터 검증 ({endpoint})")
    print("-" * 50)
    for q in test_quarters:
        rows = fetch_all(endpoint, extra=q, max_rows=1000)
        df_test = pd.DataFrame(rows)
        if not df_test.empty and 'STDR_YYQU_CD' in df_test.columns:
            actual_q = df_test['STDR_YYQU_CD'].unique()
            print(f"요청: {q} | 응답 행수: {len(df_test)}행 | 실제 분기값: {actual_q}")
        else:
            print(f"요청: {q} | 빈 응답")


[3.1 매출] 분기 필터 검증 (VwsmAdstrdSelngW)
--------------------------------------------------
요청: 20231 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20231']
Length: 1, dtype: str
요청: 20232 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20232']
Length: 1, dtype: str
요청: 20233 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20233']
Length: 1, dtype: str

[3.2 유동인구] 분기 필터 검증 (VwsmAdstrdFlpopW)
--------------------------------------------------
요청: 20231 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20251', '20252', '20241']
Length: 3, dtype: str
요청: 20232 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20251', '20252', '20241']
Length: 3, dtype: str
요청: 20233 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20251', '20252', '20241']
Length: 3, dtype: str

[3.3 점포수] 분기 필터 검증 (VwsmAdstrdStorW)
--------------------------------------------------
요청: 20231 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20231']
Length: 1, dtype: str
요청: 20232 | 응답 행수: 1000행 | 실제 분기값: <StringArray>
['20232']
Length: 1, dtype: str
요청: 20233 | 응답 행수: 1000행 | 실제

→ 3.2 유동인구 & 3.5 집객시설 & 3.6 변화지표 분기 필터 무시 확인, 정적 처리 예정

#### 분기별 수집

In [ ]:
print("3.1 VwsmAdstrdSelngW 행정동·업종·분기 매출 수집")
frames = []
for q in QUARTERS:
    rows = fetch_all('VwsmAdstrdSelngW', extra=q)
    if rows:
        df = pd.DataFrame(rows)
        df['q'] = q
        frames.append(df)
        print(f'{q}: {len(df)}행')
    else:
        print(f'{q}: 빈 응답')

df_sales = pd.concat(frames, ignore_index=True)
print(f'\n총 수집 행수: {len(df_sales)}')
print(f'컬럼: {list(df_sales.columns)[:10]}')

3.1 VwsmAdstrdSelngW 행정동·업종·분기 매출 수집
20201: 16502행
20202: 16648행
20203: 16676행
20204: 16675행
20211: 17536행
20212: 17541행
20213: 17514행
20214: 17480행
20221: 17394행
20222: 17401행
20223: 17346행
20224: 17299행
20231: 17188행
20232: 17196행
20233: 17137행
20234: 17122행
20241: 17044행
20242: 17048행
20243: 16937행
20244: 16871행
20251: 16770행
20252: 16794행
20253: 16831행
20254: 16718행

총 수집 행수: 409668
컬럼: ['STDR_YYQU_CD', 'ADSTRD_CD', 'ADSTRD_CD_NM', 'SVC_INDUTY_CD', 'SVC_INDUTY_CD_NM', 'THSMON_SELNG_AMT', 'THSMON_SELNG_CO', 'MDWK_SELNG_AMT', 'WKEND_SELNG_AMT', 'MON_SELNG_AMT']


In [14]:
print("3.3 VwsmAdstrdStorW 행정동·업종·분기 점포수 수집")
frames = []
for q in QUARTERS:
    rows = fetch_all('VwsmAdstrdStorW', extra=q)
    if rows:
        df = pd.DataFrame(rows)
        df['q'] = q
        frames.append(df)
        print(f'{q}: {len(df)}행')
    else:
        print(f'{q}: 빈 응답')

df_stores = pd.concat(frames, ignore_index=True)
print(f'\n총 수집 행수: {len(df_stores)}')
print(f'컬럼: {list(df_stores.columns)[:10]}')

3.3 VwsmAdstrdStorW 행정동·업종·분기 점포수 수집
20201: 34938행
20202: 34975행
20203: 36584행
20204: 35114행
20211: 35105행
20212: 35088행
20213: 35081행
20214: 35121행
20221: 35124행
20222: 35129행
20223: 35160행
20224: 35208행
20231: 35261행
20232: 35282행
20233: 35331행
20234: 35361행
20241: 35330행
20242: 35267행
20243: 35207행
20244: 35197행
20251: 35281행
20252: 35294행
20253: 35326행
20254: 35317행

총 수집 행수: 846081
컬럼: ['STDR_YYQU_CD', 'ADSTRD_CD', 'ADSTRD_CD_NM', 'SVC_INDUTY_CD', 'SVC_INDUTY_CD_NM', 'STOR_CO', 'SIMILR_INDUTY_STOR_CO', 'OPBIZ_RT', 'OPBIZ_STOR_CO', 'CLSBIZ_RT']


In [15]:
print("3.4 VwsmAdstrdWrcPopltnW 행정동·분기 직장인구 수집")
frames = []
for q in QUARTERS:
    rows = fetch_all('VwsmAdstrdWrcPopltnW', extra=q)
    if rows:
        df = pd.DataFrame(rows)
        df['q'] = q
        frames.append(df)
        print(f'{q}: {len(df)}행')
    else:
        print(f'{q}: 빈 응답')

df_workers = pd.concat(frames, ignore_index=True)
print(f'\n총 수집 행수: {len(df_wrc)}')
print(f'컬럼: {list(df_wrc.columns)[:10]}')

3.4 VwsmAdstrdWrcPopltnW 행정동·분기 직장인구 수집
20201: 414행
20202: 414행
20203: 414행
20204: 414행
20211: 414행
20212: 414행
20213: 414행
20214: 414행
20221: 414행
20222: 414행
20223: 414행
20224: 414행
20231: 414행
20232: 414행
20233: 414행
20234: 414행
20241: 414행
20242: 414행
20243: 414행
20244: 414행
20251: 414행
20252: 414행
20253: 414행
20254: 414행

총 수집 행수: 9936
컬럼: ['STDR_YYQU_CD', 'ADSTRD_CD', 'ADSTRD_CD_NM', 'TOT_WRC_POPLTN_CO', 'ML_WRC_POPLTN_CO', 'FML_WRC_POPLTN_CO', 'AGRDE_10_WRC_POPLTN_CO', 'AGRDE_20_WRC_POPLTN_CO', 'AGRDE_30_WRC_POPLTN_CO', 'AGRDE_40_WRC_POPLTN_CO']


#### 정적 처리 수집

In [16]:
print('3.2 VwsmAdstrdFlpopW 행정동·분기 유동인구 수집')
rows = fetch_all('VwsmAdstrdFlpopW')
df_flpop = pd.DataFrame(rows)
latest_q = df_flpop['STDR_YYQU_CD'].max()
df_flpop_static = df_flpop[df_flpop['STDR_YYQU_CD'] == latest_q].drop_duplicates(subset=['ADSTRD_CD'])
print(f'→ 최신 분기: {latest_q}, 행정동 수: {len(df_flpop_static)}개')

3.2 VwsmAdstrdFlpopW 행정동·분기 유동인구 수집
→ 최신 분기: 20254, 행정동 수: 425개


In [17]:
print("3.5 VwsmAdstrdFcltyW 행정동·분기 집객시설 수집 및 정적 처리")
rows = fetch_all('VwsmAdstrdFcltyW')
df_fclty = pd.DataFrame(rows)
latest_q = df_fclty['STDR_YYQU_CD'].max()
df_fclty_static = df_fclty[df_fclty['STDR_YYQU_CD'] == latest_q].drop_duplicates(subset=['ADSTRD_CD'])
print(f'→ 최신 분기: {latest_q}, 행정동 수: {len(df_fclty_static)}개')

3.5 VwsmAdstrdFcltyW 행정동·분기 집객시설 수집 및 정적 처리
→ 최신 분기: 20254, 행정동 수: 425개


In [18]:
print("3.6 VwsmAdstrdIxQq 행정동·분기 변화지표 수집 및 정적 처리")
rows = fetch_all('VwsmAdstrdIxQq')
df_ix = pd.DataFrame(rows)
latest_q = df_ix['STDR_YYQU_CD'].max()
df_ix_static = df_ix[df_ix['STDR_YYQU_CD'] == latest_q].drop_duplicates(subset=['ADSTRD_CD'])
print(f'→ 최신 분기: {latest_q}, 행정동 수: {len(df_ix_static)}개')

3.6 VwsmAdstrdIxQq 행정동·분기 변화지표 수집 및 정적 처리
→ 최신 분기: 20254, 행정동 수: 425개


#### csv 파일 저장

In [19]:
df_sales.to_csv(RAW_DIR / 'team3_adstrd_selng.csv', encoding='utf-8-sig', index=False)
print(f'3.1 매출 저장 완료: {len(df_sales):,}행')

df_flpop_static.to_csv(RAW_DIR / 'team3_adstrd_flpop_static.csv', encoding='utf-8-sig', index=False)
print(f'3.2 유동인구 저장 완료: {len(df_flpop_static)}행')

df_stores.to_csv(RAW_DIR / 'team3_adstrd_stor.csv', encoding='utf-8-sig', index=False)
print(f'3.3 점포수 저장 완료: {len(df_stores):,}행')

df_wrc.to_csv(RAW_DIR / 'team3_adstrd_wrc_popltn.csv', encoding='utf-8-sig', index=False)
print(f'3.4 직장인구 저장 완료: {len(df_wrc):,}행')

df_fclty_static.to_csv(RAW_DIR / 'team3_adstrd_fclty_static.csv', encoding='utf-8-sig', index=False)
print(f'3.5 집객시설 저장 완료: {len(df_fclty_static)}행')

df_ix_static.to_csv(RAW_DIR / 'team3_adstrd_ix_static.csv', encoding='utf-8-sig', index=False)
print(f'3.6 변화지표 저장 완료: {len(df_ix_static)}행')

3.1 매출 저장 완료: 409,668행
3.2 유동인구 저장 완료: 425행
3.3 점포수 저장 완료: 846,081행
3.4 직장인구 저장 완료: 9,936행
3.5 집객시설 저장 완료: 425행
3.6 변화지표 저장 완료: 425행


#### 검증 체크리스트


In [20]:
# 검증 1: 행정동 매출 데이터 있는 곳 비율 > 60%
total_adstrd = df_map['ADSTRD_CD'].nunique()
sales_adstrd = df_sales['ADSTRD_CD'].nunique()
ratio = sales_adstrd / total_adstrd * 100
print(f'[검증1] 매출 데이터 있는 행정동 비율: {ratio:.1f}% (기준: > 60%)')

# 검증 2: 자치구별 행정동 수 매트릭스
print('\n[검증2] 자치구별 행정동 수:')
gu_adstrd = df_map.groupby('SIGNGU_CD_NM')['ADSTRD_CD_NM'].nunique().sort_values(ascending=False)
print(gu_adstrd.to_string())

# 검증 3: 분기당 평균 행 수 (정상: 약 4,200행/분기)
avg_rows = df_sales.groupby('STDR_YYQU_CD').size().mean()
print(f'[검증3] 분기당 평균 행 수: {avg_rows:.0f}행 (기준: 약 4,200행)')


[검증1] 매출 데이터 있는 행정동 비율: 106.5% (기준: > 60%)

[검증2] 자치구별 행정동 수:
SIGNGU_CD_NM
관악구     21
송파구     21
강남구     20
강서구     19
성북구     18
영등포구    18
종로구     17
강동구     16
양천구     16
용산구     16
은평구     16
중랑구     16
성동구     16
서초구     16
마포구     16
광진구     15
노원구     15
중구      15
동작구     15
구로구     14
동대문구    14
도봉구     14
강북구     13
서대문구    13
금천구     10
[검증3] 분기당 평균 행 수: 17070행 (기준: 약 4,200행)


#### 검증 결과 보고
검증1) 매출 데이터 있는 행정동 비율: 106.5%
→ 매핑 파일(399개) 대비 매출 API 행정동(414개)이 더 많아 100% 초과. 데이터 이상 없음.

검증2) 자치구별 행정동 수: 강남구 20동, 강북구 13동 등 실제 행정동 수와 일치. 정상.

검증3) 분기당 평균 행 수: 17,070행
→ 매출 API가 행정동 × 업종 단위라 4,200행 기준 대비 높게 나옴. 정상.

#### 행정동·업종·분기 단위 통합 테이블 


In [10]:
# 매출 데이터 로드
df_sales = pd.read_csv(RAW_DIR / 'team3_adstrd_selng.csv', encoding='utf-8-sig', dtype={'ADSTRD_CD': str})

# 매핑 파일로 자치구 붙이기
df_map = pd.read_csv(RAW_DIR / 'TbgisTrdarRelm.csv', dtype=str)
mapping = df_map[['ADSTRD_CD', 'SIGNGU_CD_NM']].drop_duplicates()

base = df_sales.merge(mapping, on='ADSTRD_CD', how='left')
base = base.rename(columns={'SIGNGU_CD_NM': 'gu'})

print(f'기본 데이터셋 생성 완료: {len(base):,}행')
print(f'자치구 매핑 비율: {base["gu"].notna().mean()*100:.1f}%')

기본 데이터셋 생성 완료: 410,771행
자치구 매핑 비율: 96.2%


In [11]:
# 행정동별 매출 데이터 비율 확인
total_adstrd = base['ADSTRD_CD'].nunique()
print(f'행정동 수: {total_adstrd}개')

# 자치구별 행정동 수 매트릭스
print('\n자치구별 행정동 수:')
print(base.groupby('gu')['ADSTRD_CD'].nunique().sort_values(ascending=False))

# 결측 비율 확인
missing_ratio = base['gu'].isna().mean() * 100
print(f'\n자치구 매핑 결측 비율: {missing_ratio:.1f}%')

if missing_ratio > 50:
    print('Fallback 필요: 매출 상위 N개 동만 사용')
else:
    print('정상: 통합 진행 가능')

행정동 수: 425개

자치구별 행정동 수:
gu
관악구     21
송파구     21
강남구     20
강서구     19
성북구     18
영등포구    18
종로구     17
강동구     16
양천구     16
용산구     16
은평구     16
중랑구     16
성동구     16
서초구     16
마포구     16
광진구     15
노원구     15
중구      15
동작구     15
구로구     14
동대문구    14
도봉구     14
강북구     13
서대문구    13
금천구     10
Name: ADSTRD_CD, dtype: int64

자치구 매핑 결측 비율: 3.8%
정상: 통합 진행 가능


In [12]:
# 3.3 점포수 결합 (행정동 + 업종 + 분기 기준)
df_stor = pd.read_csv(RAW_DIR / 'team3_adstrd_stor.csv', encoding='utf-8-sig', dtype={'ADSTRD_CD': str})
stor_cols = ['ADSTRD_CD', 'SVC_INDUTY_CD', 'STDR_YYQU_CD', 'STOR_CO', 'OPBIZ_RT', 'CLSBIZ_RT']
base = base.merge(df_stor[stor_cols], on=['ADSTRD_CD', 'SVC_INDUTY_CD', 'STDR_YYQU_CD'], how='left')
print(f'점포수 결합 후: {len(base):,}행')

# 3.4 직장인구 결합 (행정동 + 분기 기준)
df_wrc = pd.read_csv(RAW_DIR / 'team3_adstrd_wrc_popltn.csv', encoding='utf-8-sig', dtype={'ADSTRD_CD': str})
wrc_cols = ['ADSTRD_CD', 'STDR_YYQU_CD', 'TOT_WRC_POPLTN_CO']
base = base.merge(df_wrc[wrc_cols], on=['ADSTRD_CD', 'STDR_YYQU_CD'], how='left')
print(f'직장인구 결합 후: {len(base):,}행')

# 3.2 유동인구 결합 (정적 — 행정동 기준만)
df_flpop = pd.read_csv(RAW_DIR / 'team3_adstrd_flpop_static.csv', encoding='utf-8-sig', dtype={'ADSTRD_CD': str})
flpop_cols = ['ADSTRD_CD', 'TOT_FLPOP_CO']
base = base.merge(df_flpop[flpop_cols], on='ADSTRD_CD', how='left')
print(f'유동인구 결합 후: {len(base):,}행')

# 3.5 집객시설 결합 (정적 — 행정동 기준만)
df_fclty = pd.read_csv(RAW_DIR / 'team3_adstrd_fclty_static.csv', encoding='utf-8-sig', dtype={'ADSTRD_CD': str})
base = base.merge(df_fclty, on='ADSTRD_CD', how='left', suffixes=('', '_fclty'))
print(f'집객시설 결합 후: {len(base):,}행')

# 3.6 변화지표 결합 (정적 — 행정동 기준만)
df_ix = pd.read_csv(RAW_DIR / 'team3_adstrd_ix_static.csv', encoding='utf-8-sig', dtype={'ADSTRD_CD': str})
ix_cols = ['ADSTRD_CD', 'TRDAR_CHNGE_IX', 'TRDAR_CHNGE_IX_NM', 'OPR_SALE_MT_AVRG', 'CLS_SALE_MT_AVRG']
base = base.merge(df_ix[ix_cols], on='ADSTRD_CD', how='left')
print(f'변화지표 결합 후: {len(base):,}행')

점포수 결합 후: 410,771행
직장인구 결합 후: 410,771행
유동인구 결합 후: 410,771행
집객시설 결합 후: 410,771행
변화지표 결합 후: 410,771행


In [13]:
# 수집 시 생성했던 기존 'q' 파생 변수가 원본 데이터에 남아있다면 삭제
# (단, STDR_YYQU_CD가 아직 존재할 때만 삭제하여 중복 실행 에러 방지)
if 'q' in base.columns and 'STDR_YYQU_CD' in base.columns:
    base = base.drop(columns=['q'])
    
# 협업 규칙 컬럼명으로 변경
base = base.rename(columns={
    'THSMON_SELNG_AMT':   'adstrd_sales_amt',
    'THSMON_SELNG_CO':    'adstrd_sales_count',
    'STOR_CO':            'adstrd_stor_co',
    'OPBIZ_RT':           'adstrd_opbiz_rt',
    'CLSBIZ_RT':          'adstrd_clsbiz_rt',
    'TOT_WRC_POPLTN_CO':  'adstrd_wrc_co',
    'TOT_FLPOP_CO':       'adstrd_flpop',
    'ADSTRD_CD':          'adstrd_code',
    'ADSTRD_CD_NM':       'adstrd_nm',
    'SVC_INDUTY_CD_NM':   'biz',
    'STDR_YYQU_CD':       'q',
})

# 결합키 컬럼을 맨 앞으로 정렬
key_cols = ['adstrd_code', 'adstrd_nm', 'gu', 'biz', 'q']
other_cols = [c for c in base.columns if c not in key_cols]
base = base[key_cols + other_cols]

# 최종 검증
print(f'총 행 수: {len(base):,}행')
print(f'분기당 평균 행 수: {base.groupby("q").size().mean():.0f}행')
print(f'컬럼 수: {len(base.columns)}개')
print(f'결측 비율:\n{base[key_cols].isna().mean().round(3)}')

# 최종 산출물 저장
base.to_csv(EXT_DIR / 'team3_adstrd_integrated.csv', encoding='utf-8-sig', index=False)
print('\n team3_adstrd_integrated.csv 저장 완료')

총 행 수: 410,771행
분기당 평균 행 수: 17115행
컬럼 수: 85개
결측 비율:
adstrd_code    0.000
adstrd_nm      0.000
gu             0.038
biz            0.000
q              0.000
dtype: float64

 team3_adstrd_integrated.csv 저장 완료


#### 통합 데이터셋 검증 및 후처리
병합 직후 데이터가 약 41만 행으로 확인 → 프로젝트 기준(약 10만 행)에 맞추기 위한 최종 정제 진행

- 타겟 업종 10개만 필터링하여 불필요한 업종 데이터 제거
- 자치구(gu) 정보가 없는 결측 데이터 제거하여 추후 결합 오류 방지

In [6]:
df_final = pd.read_csv(EXT_DIR / 'team3_adstrd_integrated.csv')

# 프로젝트 타겟 업종(10개) 필터링
target_biz = [
    '한식음식점', '커피-음료', '양식음식점', '호프-간이주점', '반찬가게', 
    '편의점', '슈퍼마켓', '의약품', '일반의류', '컴퓨터및주변장치판매'
]
df_final = df_final[df_final['biz'].isin(target_biz)].copy()

# 자치구(gu) 매핑이 안 된 3.8%의 결측치 행 버리기 (Fallback 처리)
df_final = df_final.dropna(subset=['gu'])

# 정제 후 최종 검증
print("\n")
print("[최종 후처리 완료 지표]")
print(f"총 행 수: {len(df_final):,}행")
print(f"분기당 평균 행 수: {df_final.groupby('q').size().mean():.0f}행")
print(f"컬럼 수: {len(df_final.columns)}개")
print(f"결측 비율 (gu): {df_final['gu'].isna().mean():.3f}")
print("\n")

# 최종본 덮어쓰기 저장
out_path = EXT_DIR / 'team3_adstrd_integrated.csv'
df_final.to_csv(out_path, encoding='utf-8-sig', index=False)
print("team3_adstrd_integrated.csv 정제 및 최종 덮어쓰기 완료")



[최종 후처리 완료 지표]
총 행 수: 82,053행
분기당 평균 행 수: 3419행
컬럼 수: 85개
결측 비율 (gu): 0.000


team3_adstrd_integrated.csv 정제 및 최종 덮어쓰기 완료
